src/Builder/FrontendDialectTransformer.cpp

```diff
@@ -1406,21 +1411,42 @@ int ImportFrontendModelArray(const void *onnxBuffer, int size,
 }
 
 // Return 0 on success, error otherwise.
-int ImportFrontendModelFile(std::string model_fname, MLIRContext &context,
+int ImportFrontendModelFile(StringRef model_fname, MLIRContext &context,
     OwningOpRef<ModuleOp> &module, std::string *errorMessage,
     ImportOptions options) {
   onnx::ModelProto model;
-  std::fstream input(model_fname, std::ios::in | std::ios::binary);
-  // check if the input file is opened
-  if (!input.is_open()) {
-    *errorMessage = "Unable to open or access " + model_fname;
-    return InvalidInputFileAccess;
-  }
+  if (model_fname.endswith(".json")) {
+    auto buf = openInputFile(model_fname, errorMessage);
+    if (!buf) {
+      return InvalidInputFileAccess;
+    }
+    // Remove // comments, which are non-standard json but appear in lit tests
```

Comment:
Lines are removed if the start with `//` ? 
Maybe an error can be returned if you detect JSON lines that have a `//` in them (aka comments that terminate the json line).
Ignore if difficult to do, but maybe be explicit in the comments here on which comments we tolerate and which one we do not (if applicable)

Reply:
yes, lines starting with `//` are ignored now

all the relevant RUN/COM/CHECK/FAILED commands in lit tests appear in lines starting with `//`

end-of-line // comments are very rare in lit tests and I suggest that we don't support them in the parse lit tests

I added a warning printed to stderr if there's a // in a line that doesn't begin with //, this way if you put an end-of-line comment in a json file you will see the warning as additional diagnostic info to go with the json parsing error (or, if the // was part of some valid json you can ignore the false-positive warning)